name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/FinnhubSymbolSearch.java

```diff
@@ -0,0 +1,170 @@
+package name.abuchen.portfolio.online.impl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.JSONValue;
+
+import name.abuchen.portfolio.PortfolioLog;
+import name.abuchen.portfolio.model.ClientSettings;
+import name.abuchen.portfolio.model.Security;
+import name.abuchen.portfolio.online.SecuritySearchProvider.ResultItem;
+import name.abuchen.portfolio.util.WebAccess;
+import name.abuchen.portfolio.util.WebAccess.WebAccessException;
+
+/**
+ * Use the <a href="https://finnhub.io/docs/api/symbol-search">Finnhub Symbol Lookup</a> API 
+ * to search for securities by symbol or ISIN. This API can be used with the free API key that
+ * you can obtain by registering on <a href="https://finnhub.io/>FinnHub.io</a>.
+ * 
+ * @see FinnhubSearchProvider
+ */
+public class FinnhubSymbolSearch
+{
+    private String apiKey;
+
+    public FinnhubSymbolSearch(String apiKey)
+    {
+        this.apiKey = apiKey;
+    }
+
+    public void setApiKey(String apiKey)
```

Comment:
I don't think that method is used (as the API key is only set via the constructor).